MediaProcessor/src/FFmpegCommandBuilder.cpp

```diff
@@ -0,0 +1,96 @@
+#include "FFmpegCommandBuilder.h"
+
+#include "ConfigManager.h"
+#include "FFmpegSettingsManager.h"
+#include "Utils.h"
+
+namespace MediaProcessor {
+
+FFmpegCommandBuilder::FFmpegCommandBuilder(FFmpegSettingsManager& ffmpegSettings)
+    : m_ffmpegSettings(ffmpegSettings),
+      m_ffmpegPath(ConfigManager::getInstance().getFFmpegPath()) {}
+
+FFmpegCommandBuilder& FFmpegCommandBuilder::addOverwrite() {
+    addFlag("-y");
+
+    return *this;
+}
+
+FFmpegCommandBuilder& FFmpegCommandBuilder::addInputFile(const fs::path& inputFile) {
+    m_inputFile = inputFile;
+    addFlag("-i", inputFile.string());
+
+    return *this;
+}
+
+FFmpegCommandBuilder& FFmpegCommandBuilder::addOutputFile(const fs::path& outputFile) {
+    m_outputFile = outputFile;
+    addArgument(outputFile.string());
+
+    return *this;
+}
+
+FFmpegCommandBuilder& FFmpegCommandBuilder::addAudioCodec() {
+    std::string codec = Utils::enumToString<AudioCodec>(m_ffmpegSettings.getAudioCodec(),
+                                                        m_ffmpegSettings.getAudioCodecToString());
+
+    if (codec.compare("unknown") == 0) {
+        throw std::runtime_error("Audio codec is unknown.");
+    }
+
+    addFlag("-c:a", codec);
+
+    return *this;
+}
+
+FFmpegCommandBuilder& FFmpegCommandBuilder::addAudioSampleRate() {
+    int sampleRate = m_ffmpegSettings.getAudioSampleRate();
+
+    // Nyquist-Shannon theorem
+    if (sampleRate <= 40000) {
```

Comment:
I'll explain in the review comment, but let's remove checks done on setting fields. Anything that doesn't come from the settings manager, we should still check.